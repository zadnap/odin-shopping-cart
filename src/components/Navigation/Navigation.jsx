import { faHouse } from '@fortawesome/free-solid-svg-icons/faHouse';
import { faCalendar, faFire, faHeart } from '@fortawesome/free-solid-svg-icons';
import NavItem from '../NavItem/NavItem';
import styles from './Navigation.module.scss';

function Navigation() {
  const navList = [
    {
      to: '/',
      icon: faHouse,
      title: 'Home',
    },
    {
      to: '/favourtites',
      icon: faHeart,
      title: 'Favourites',
    },
    {
      to: '/upcoming',
      icon: faCalendar,
      title: 'Coming Soon',
    },
    {
      to: '/trending',
      icon: faFire,
      title: 'Trending',
    },
  ];

  return (
    <nav className={styles.navigation}>
      {navList.map((item) => (
        <li key={item.to}>
          <NavItem icon={item.icon} to={item.to}>
            {item.title}
          </NavItem>
        </li>
      ))}
    </nav>
  );
}

export default Navigation;
