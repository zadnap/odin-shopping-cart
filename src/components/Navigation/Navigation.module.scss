@use '@/styles/variables.scss' as *;

.navigation {
  position: fixed;
  background-color: $background-color;
  width: 250px;
}
