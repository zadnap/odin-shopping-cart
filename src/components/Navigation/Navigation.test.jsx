import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navigation from './Navigation';

describe('Navigation component', () => {
  it('should render to match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
