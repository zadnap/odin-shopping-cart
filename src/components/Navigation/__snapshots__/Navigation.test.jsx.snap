// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation component > should render to match snapshot 1`] = `
<div>
  <nav
    class="_navigation_0efb96"
  >
    <li>
      <a
        class="_navItem_a6b845 _active_a6b845"
        data-discover="true"
        href="/"
      >
        <span
          data-testid="fa-icon"
        />
        Home
      </a>
    </li>
    <li>
      <a
        class="_navItem_a6b845 "
        data-discover="true"
        href="/favourtites"
      >
        <span
          data-testid="fa-icon"
        />
        Favourites
      </a>
    </li>
    <li>
      <a
        class="_navItem_a6b845 "
        data-discover="true"
        href="/upcoming"
      >
        <span
          data-testid="fa-icon"
        />
        Coming Soon
      </a>
    </li>
    <li>
      <a
        class="_navItem_a6b845 "
        data-discover="true"
        href="/trending"
      >
        <span
          data-testid="fa-icon"
        />
        Trending
      </a>
    </li>
  </nav>
</div>
`;
